### PR TITLE
Fix problem when a variable is set to All and use repeat panels

### PR DESCRIPTION
**Pull request by leoncx**
Validate if a scoped variable is not set to All. This scoped variable
will be store in warp10 query.

Refs: #28

### CODE REVIEW

**Comment by aurrelhebert** [comment]
Hi @leoncx, thanks for your PR, we just merged one to update our dependencies which conflict with yours. Could you resolve you conflict ?

File: src/warp10-datasource.ts [hunk 1]
```diff
@@ -296,11 +296,28 @@ export default class Warp10Datasource {
...
+        if (v.selected || this.scopedVarIsAll(k)) {
           str += `'${v.value}' '${k}' STORE `
         }
       }
     }
     return str
   }
+
+  /**
+   * Test is a named scoped variable is set to all
```
Comment (miton18) [inline src/warp10-datasource.ts]: `Test if` ?

File: src/warp10-datasource.ts [hunk 2]
```diff
@@ -296,11 +296,28 @@ export default class Warp10Datasource {
...
+   *
+   * @param name string The name of scoped variable
+   * @return bool If the scoped variable is set to all
+   */
+  private scopedVarIsAll(name): boolean {
```
Comment (miton18) [inline src/warp10-datasource.ts]: You can add `name` parameter type
`private scopedVarIsAll(name: string): boolean`

File: src/warp10-datasource.ts [hunk 3]
```diff
@@ -296,11 +296,28 @@ export default class Warp10Datasource {
...
+    for (let i = 0; i < this.templateSrv.variables.length; i++) {
+      let v = this.templateSrv.variables[i]
```
Comment (miton18) [inline src/warp10-datasource.ts]: Use `const` instead of `let` (because v is not altered)

File: src/warp10-datasource.ts [hunk 4]
```diff
@@ -296,11 +296,28 @@ export default class Warp10Datasource {
...
+      if (v.name === name && v.current.value === '$__all') {
```
Comment (miton18) [inline src/warp10-datasource.ts:316]: Tips: when you look for only one element in an array matching a condition, you can use [Array.find()](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/find)

**Review by miton18 — CHANGES_REQUESTED** [review]

File: src/warp10-datasource.ts (hunk 4 above)
Comment (leoncx) [reply src/warp10-datasource.ts:316]: I pushed a new version with the modifications based on comments. I will look at this one later.

**Review by aurrelhebert — APPROVED** [review]

**Review by miton18 — APPROVED** [review]

**Pull request merged**